components/camel-cdi/src/main/java/org/apache/camel/cdi/CdiCamelExtension.java

```diff
@@ -244,6 +253,22 @@ private void afterBeanDiscovery(@Observes AfterBeanDiscovery abd, BeanManager ma
         // The set of extra Camel CDI beans
         Set<SyntheticBean<?>> extraBeans = new HashSet<>();
 
+        // Find DefaultContextCreateStrategy
+        Bean<?> defaultContextCreateStrategy = cdiBeans.stream()
+            .filter(hasType(DefaultContextCreateStrategy.class))
+            .findFirst()
+            .orElse(null);
+
+        if (defaultContextCreateStrategy == null) {
+            abd.addDefinitionError(new IllegalStateException("Unable to find DefaultContextCreateStrategy bean"));
+        }
+
+        // Use custom ContextCreateStrategy if one exists
+        contextCreateStrategy = cdiBeans.stream()
```

Comment:
The logic of resolving `ContextCreateStrategy` could be encapsulated in `ContextCreateStrategyInjectionTarget` and the corresponding field removed.